atlasdb-client/src/main/java/com/palantir/atlasdb/tracing/TraceStatistic.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.tracing;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public final class TraceStatistic {
+    private static final TraceStatistic NOT_OBSERVED_TRACE = empty();
+
+    private final AtomicLong emptyReads;
+    private final AtomicLong skippedValues;
+    private final AtomicLong bytesReadFromDb;
+
+    private TraceStatistic(long emptyReads, long skippedValues, long bytesReadFromDb) {
+        this.emptyReads = new AtomicLong(emptyReads);
+        this.skippedValues = new AtomicLong(skippedValues);
+        this.bytesReadFromDb = new AtomicLong(bytesReadFromDb);
```

Comment:
Do we end up mutating these from multiple threads concurrently? If so, we might want to consider [LongAdder](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/atomic/LongAdder.html) for these

Reply:
Good call; it can be mutated from multiple threads when a lazy call is handled by multiple threads